py3status/core.py

```diff
@@ -326,18 +346,13 @@ def timeout_process_add_queue(self, module, cache_time):
             if cache_time not in self.timeout_keys:
                 self.timeout_queue[cache_time] = {module}
                 self.timeout_keys.append(cache_time)
-                # sort keys so earliest is first
-                self.timeout_keys.sort()
 
-                # when is next timeout due?
-                try:
-                    self.timeout_due = self.timeout_keys[0]
-                except IndexError:
-                    self.timeout_due = None
+                self._set_new_timeout_due()
             else:
                 self.timeout_queue[cache_time].add(module)
             # note that the module is in the timeout_queue
             self.timeout_queue_lookup[module] = cache_time
+            self.timeout_queue_lookup_previous[module] = cache_time
```

Comment:
Looks to me that `self.timeout_queue_lookup_previous[module]` should also be cleared when `self.timeout_queue_lookup[module]` and not ONLY when `cache_time == Py3.CACHE_FOREVER:` ; else this dict will only grow in memory

Reply:
Well. We can clear it after we use it's value otherwise there will be no pointer for detecting ongoing timeout_due for module. Or there is other possibilities to overcome that issue?
Did you test your ideas by yourself or you just shared feeling?

Reply:
Definetly we need hold old values because at the runtime there isn't available data at dict self.timeout_queue_lookup.
Well after opening  debugger i noticed variable timeout_running which needs some attention.

Reply:
Yes, but if the module gets processed, it can safely be dropped from timeout_queue_lookup right?

So, this could be added on L367 I guess

Reply:
Actually I'm stupid sorry, the key is the module so this won't pile up ...